### PR TITLE
Change equal for path element

### PR DESCRIPTION
**Pull request by apelisse**
This brings a 5 to 10% time-wise on some of the operations. Might be tremendous improvements on some specific resources though.

### CODE REVIEW

File: fieldpath/element.go [hunk 1]
```diff
@@ -49,50 +49,90 @@ type PathElement struct {
...
 	}
 
-	return false
+	return 0
 }
 
 // Equals returns true if both path elements are equal.
 func (e PathElement) Equals(rhs PathElement) bool {
-	return !e.Less(rhs) && !rhs.Less(e)
+	if e.FieldName != nil {
```
Comment (jpbetz) [inline fieldpath/element.go]: For readability, could use this idiom for taking care of the nil cases up-front:
```
if (e.FieldName == nil) != (rhs.FieldName == nil) {
  return false
}
```

Just a suggestion. The logic already appears correct.

Comment (apelisse) [reply fieldpath/element.go]: Maybe it's just me, but I had to squint to make sure that statement was correct, I somehow find it harder to read?

**Comment by apelisse** [comment]
yay

**Pull request merged**